src/transformStateWithClones.js

```diff
@@ -7,7 +7,38 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const copy = [];
+  let copyObj = { ...state };
+
+  for (const action of actions) {
+    const type = action.type;
+
+    switch (type) {
+      case 'addProperties':
+        Object.assign(copyObj, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const keys of action.keysToRemove) {
+          if (copyObj.hasOwnProperty(keys)) {
+            delete copyObj[keys];
+          }
+        }
+        break;
+
+      case 'clear':
+        copyObj = {};
+        break;
+
+      default:
+        copy.push({ ...copyObj });
```

Comment:
Better just break or throw some error, because now you will push it twice (here and on 38 line)